maven/build.go

```diff
@@ -97,6 +98,11 @@ func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
 		return libcnb.BuildResult{}, fmt.Errorf("unable to resolve build arguments\n%w", err)
 	}
 
+	if !isatty.IsTerminal(os.Stdout.Fd()) && !contains(args, []string{"-B", "--batch-mode"}) {
```

Comment:
When I suggested we check whether stdout is a terminal I forgot a bit of complexity.

In theory `Stdout` for an executed command isn't guaranteed to be the same as `os.Stdout`. The writer returned by `c.Logger.InfoWriter()` will eventually be used to set `Stdout` on the command. First the [logger is passed to the Application](https://github.com/paketo-buildpacks/maven/blob/main/maven/build.go#L129). Then, the [logger writer is passed to the command](https://github.com/paketo-buildpacks/libbs/blob/main/application.go#L55)

In practice this value is always `os.Stdout` because we [construct the logger from `os.Stdout`](https://github.com/paketo-buildpacks/maven/blob/main/cmd/main/main.go#L33) in `main.go` and inject it into `Build`.

For now, I think we should detect whether `os.Stdout` is a tty in `main.go` and add a `TTY` field with type bool on the build struct. This has the additional benefit of allowing us to inject test values, enabling us to unit test all cases (regardless of whether the unit tests are running in a tty).

Another option would be to add another function like `IsATTY` to the logger interface. But this is more work and would touch a couple repos. I suggest we take the easier path for this change.

Reply:
> For now, I think we should detect whether os.Stdout is a tty in main.go and add a TTY field with type bool on the build struct.

Agreed. I made the required changes.
Let me know!
Thanks